### PR TITLE
Add access control

### PR DESCRIPTION
**Pull request by niansa**
This pull request adds a *Access Control List* that allows server admins to control which roles are allowed to use the bot and which are not.

Preview:
grafik

Fixes https://github.com/Magic-Bots/xenon/issues/97

### CODE REVIEW

**Comment by niansa** [comment]
Finally, it's ready for review!!!

**Review by merlinfuchs — COMMENTED** [review]
I'm still not sure about this. I will have a more specific look on it when I get time.

**Comment by niansa** [comment]
:thinking:

**Comment by niansa** [comment]
Alright

**Comment by merlinfuchs** [comment]
@ThatLifeofGames Do you think this a good thing to add? Or would a switch between owner only and all admins be enough / better. 
And what do you think about the idiot test?

**Comment by ThatLifeofGames** [comment]
Yeah I feel like people will be stupid with this, give access to everyone and complain at us for it. A simple toggle between admin and owner would probably be best.

**Comment by ThatLifeofGames** [comment]
However, this could be re-used to make a full settings panel with more configuration options.

**Comment by niansa** [comment]
What about this:
Roles without the privilege to manage channels can't have access to the bot.
?

File: xenon/betterdiscordctl [hunk 1]
```diff
@@ -0,0 +1,646 @@
+#!/usr/bin/env bash
+
+set -ueo pipefail
+shopt -s dotglob extglob nullglob
+
+# Constants
+VERSION=1.6.1
+SOURCE=$(readlink -f "${BASH_SOURCE[0]}")
+GITHUB_URL='https://raw.githubusercontent.com/bb010g/betterdiscordctl/master/betterdiscordctl'
```
Comment (niansa) [inline xenon/betterdiscordctl]: Oops, betterdiscordctl isn't right in place here. For some reason it was inside my local repository...

**Comment by niansa** [comment]
Okay I can't find any more bugs. :D
So ready for next review

**Comment by niansa** [comment]
I am thinking about converting this to an entire settings menu... What do you think? Or should I do that in another pull request?

**Comment by niansa** [comment]
Reason? Wow

**Comment by merlinfuchs** [comment]
Sorry, I recreated the dev branch. You need to update the branch (idk if that's possible) or make a new pull request.

**Comment by niansa** [comment]
Oh, alright, already deleted the sources but I still have them mirrored to my server. I'll recreate as soon as possible.

**Pull request closed without merging**